work12/index.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+const wrap = document.getElementById("js-wrap")
+const ul = document.getElementById("js-list");
+const button = document.getElementById("js-button");
+// const url = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// const url = "https://myjson.dit.upm.es/api/bins/bu5z";
+const url = "https://myjson.dit.upm.es/api/bins/2hj3";
+
+async function getData() {
+  try {
+    const response = await fetch(url);
+    if (response.ok) {
+      const json = await response.json();
+      return json;
+    } else {
+      throw new Error(response.statusText);
+    }
+  } catch (e) {
+    throw new Error(e);
+  }
+}
+
+async function getListData() {
+  loading();
+  let listData;
+  try {
```

Comment:
@yuka830 

In the code before the change, the function name and the processing content of the function did not match.
this time,
Instead of changing the function name to "init", I changed listData to the return value.
but,
```
button.addEventListener（ "click"、async（）=> {

const data = await getListData（）;

}）
```
It is redundant like.
I tried it, but as Yuka said, it seems better to change the function name.

I have a question.
Do I need to change the function name from "init" to "getListData ()"?

Reply:
> Do I need to change the function name from "init" to "getListData ()"?
Sorry, I didn't get it this part well.  You meant to change the name `getListData` → `init` in the code you showed above? 

If I were you, I would implement like this.
```
const button = document.getElementById("js-button"); ← Call it in the first part in the global

const init = async () => {
  loading();
  const data = await getListData();
  renderListElement(data);
  hideLoading();
};

button.addEventListener("click", async () => {
  init();
  button.style.display = "none";
});
```

Reply:
May be my mistake

"const button = document.getElementById (" js-button "); ← Call it in the first part in the global"

 In "Call it in the first part in the global"
I got an error.
It seems that the order of DOM generation is the cause. I thought I had to write it after firing renderButtonElement ().
Therefore, it is the content of the following commit.

commit: [459d307](https://github.com/yasu-pro/js-handson/pull/12/commits/459d307ad6c1667a23a82d3824f5d764736b8c07)

Reply:
After some experimentation, I decided to enclose the click event in "DOMContentLoaded".

commit: [cac07ad](https://github.com/yasu-pro/js-handson/pull/12/commits/cac07ad5097ac60ae01e46f6024dcf08cfc08d5b)

Reply:
Oh you changed the buttons to be made with js.
Yeah, then I was wrong... I'm sorry 💦 
I think it looks good now!

Reply:
Thank you

Reply:
You don't have to say sorry, it's my mistake!
Thanks 😊